go/vt/vtgate/router_dml_test.go

```diff
@@ -477,6 +507,36 @@ func TestInsertGeneratorUnsharded(t *testing.T) {
 	}
 }
 
+func TestInsertAutoincUnsharded(t *testing.T) {
+	router, _, _, sbclookup := createRouterEnv()
+
+	// Fake a mysql auto-inc response.
+	query := "insert into simple(val) values ('val')"
+	wantResult := &sqltypes.Result{
+		Rows: [][]sqltypes.Value{{
+			sqltypes.MakeTrusted(sqltypes.Int64, []byte("1")),
+		}},
+		RowsAffected: 2,
+		InsertID:     2,
+	}
+	sbclookup.SetResults([]*sqltypes.Result{wantResult})
+
+	result, err := routerExec(router, query, nil)
+	if err != nil {
+		t.Error(err)
+	}
+	wantQueries := []querytypes.BoundQuery{{
+		Sql:           query,
+		BindVariables: map[string]interface{}{},
+	}}
+	if !reflect.DeepEqual(sbclookup.Queries, wantQueries) {
+		t.Errorf("sbclookup.Queries: \n%#v, want \n%#v\n", sbclookup.Queries, wantQueries)
+	}
+	if !reflect.DeepEqual(result, wantResult) {
```

Comment:
Must be:

`if !result.Equal(wantResult) {`

because of the upcoming protobuf changes which will break `DeepEqual`.